hs_geographic_feature_resource/forms.py

```diff
@@ -0,0 +1,254 @@
+    featureCount = forms.IntegerField(label='Feature Count', required=True, widget=forms.TextInput())
+
+    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, *args, **kwargs):
+        super(GeometryInformationForm, self).__init__(*args, **kwargs)
+        self.helper = GeometryInformationFormHelper(allow_edit, res_short_id, element_id, element_name='GeometryInformation')
+        self.delete_modal_form = None
+        self.number = 0
+        self.allow_edit = allow_edit
+        self.errors.clear()
+
+        if not allow_edit:
+            for field in self.fields.values():
+                field.widget.attrs['readonly'] = True
+                field.widget.attrs['style'] = "background-color:white;"
+
+    def clean(self):
+        # modify the form's cleaned_data dictionary
+        super(GeometryInformationForm, self).clean()
+
+        is_form_errors = False
+        out={}
+        for ele in ('geometryType', 'featureCount'):
+            if self.cleaned_data.get(ele, None):
+                out[ele]=self.cleaned_data[ele]
+            else:
+                self._errors[ele] = ["Data for %s is missing" % ele]
+                is_form_errors = True
+                break
+
+        if not is_form_errors:
+            self.cleaned_data["geometryType"] = out["geometryType"]
+            self.cleaned_data["featureCount"] = out["featureCount"]
+
+        return self.cleaned_data
+
+
+class GeometryInformationValidationForm(forms.Form):
+    featureCount = forms.IntegerField(required=True)
+    geometryType = forms.CharField(max_length=128, required=True)
+
+
+class FieldInformationFormHelper(BaseFormHelper):
+    def __init__(self, allow_edit=True, res_short_id=None, element_id=None, element_name=None,  *args, **kwargs):
+
+        # the order in which the model fields are listed for the FieldSet is the order these fields will be displayed
+        field_width = 'form-control input-sm'
+        layout = Layout(
+                       HTML(
+                        '<h4 ><strong>Field Information:</strong></h4>'
+                        '<hr style="margin-top: 0px;margin-bottom: 2px">'
+                        '<table>'
+                            '<tr class="row">'
+                                 '<th >Name</th><th class="col-sm-2">Type</th> <th class="col-sm-2">Width</th> <th class="col-sm-2">Precision</th>'
+                            '</tr>'
+                            '{%for field_item in field_information %}'
+                            'tr class="row">'
+                            '<td class="col-sm-2">{{field_item.fieldName}}</td>'
+                            '<td>{{field_item.fieldType}}</td>'
+                            '<td>{{field_item.fieldWidth}}</td>'
+                            '<td>{{field_item.fieldPrecision}}</td>'
+                            '</tr>'
+                            '{% endfor %}'
+                        '</table>'
```

Comment:
I think it would be advisable to keep HTML out of Python code. It makes it harder to track down when languages are mixed in one file. Can you move this to a template?